### ISSUE TITLE
Cannot inherit frozen dataclass from a non-frozen one error when using pydantic frozen=True

### ISSUE THREAD

**Issue by dmarteau** [astral-sh/ty#3486]
### Description

The command `ty check test.py`  with the following `test.py` file:
```python
from pydantic import BaseModel

class Base(BaseModel, frozen=True):
    pass

class Derived(Base):
   pass
```

Produce the following error:
```
error[invalid-frozen-dataclass-subclass]: Non-frozen dataclass cannot inherit from frozen dataclass
 --> test.py:6:7
  |
6 | class Derived(Base):
  |       ^^^^^^^^----^ Subclass `Derived` is not frozen but base class `Base` is
  |
info: This causes the class creation to fail

Found 1 diagnostic
```

### Result expected

No errors

### References

The error is somewhat similar to this previous [Mypy issue](https://github.com/python/mypy/issues/14857) which is resolved when using  the `pydantic.mypy` plugin.  Thus it appears to be a specific Pydantic support issue.

Thx.

### Version

ty 0.0.37

**Comment by carljm** [comment]
Yes, it looks like this is something that we'd just need to special-case for pydantic, since it behaves differently from stdlib dataclasses (and in general we are supposed to assume that `dataclass_transform` behaves like stdlib dataclass).

Just for clarity for anyone looking at this in future, the linked mypy issue is rather different -- it's about supporting a frozen subclass of `BaseModel` in the first place (something that ty and all type checkers already correctly support today), not about supporting a non-frozen subclass of a frozen model.

**Comment by sharkdp** [comment]
This is now supported, i.e. we don't raise this diagnostic for Pydantic models anymore.
